### PR TITLE
Rip out TL Master from CanHavePeripheryTLSerial trait

### PR DESCRIPTION
**Pull request by t14916**
This PR removes the TL Master from `CanHavePeripheryTLSerial`, since FireSim doesn't rely on any of the TLMaster features (i.e. the `libwif-rom` / `libwif-ram`) and uses TLSerial exclusively for tethered boot.

### CODE REVIEW

File: src/main/scala/SerialAdapter.scala [hunk 1]
```diff
@@ -97,6 +97,25 @@ case object SerialAdapter {
...
+
+    val harness = LazyModule(new SerialHarness(
+      p(SerialTLKey).get.width,
+      clientEdge = desser.node.edges.out(0)
+    ))
+
+    val module = Module(harness.module)
+    module.io.ser <> port
+
+    /*require(harness.serdes.module.mergeTypes.head.param == desser.module.mergeTypes.head.param,
```
Comment (t14916) [inline src/main/scala/SerialAdapter.scala]: in `connectHarnessRAM` there is a require here that works, I tried to match the parameters to what they are in the `desser` and `serdes` but this seems fire even when the simulation as a whole works

File: src/main/scala/Serdes.scala [hunk 2]
```diff
@@ -542,9 +542,9 @@ class TLSerdes(w: Int, params: Seq[TLManagerParameters], beatBytes: Int = 8, has
   }
 }
 
-class TLDesser(w: Int, params: Seq[TLClientParameters], hasCorruptDenied: Boolean = true)(implicit p: Parameters) extends LazyModule {
+class TLDesser(w: Int, params: Seq[TLMasterPortParameters], hasCorruptDenied: Boolean = true)(implicit p: Parameters) extends LazyModule {
```
Comment (davidbiancolin) [inline src/main/scala/Serdes.scala:545]: What's the purpose of this change. It's just a small clean up?

File: src/main/scala/SerialAdapter.scala (hunk 1 above)
Comment (davidbiancolin) [reply src/main/scala/SerialAdapter.scala]: Can you DM me the requirement?

File: src/main/scala/Serdes.scala (hunk 2 above)
Comment (t14916) [reply src/main/scala/Serdes.scala:545]: The `Serdesser` doesn't use the `TLClientParameters` or `TLManagerParameters`  because these parameters are actually deprecated. So a lot of the `SerialAdapter` code is using the `TLMasterPortParameters`/`TLSlavePortParameters`. Since both the `Desser` and `Serdes` are just converting the parameters internally it makes more sense to just directly pass them in since that's what's happening in the `Serdesser`

File: src/main/scala/SerialAdapter.scala (hunk 1 above)
Comment (t14916) [reply src/main/scala/SerialAdapter.scala]: out of band: removing this require is fine.

**Comment by andrew-gouldey-sifive** [comment]
The TL stuff looks OK to me.  I am not familiar with the testchipip repo, so I'll let @davidbiancolin approve.

**Review by davidbiancolin — APPROVED** [review]
